assets/js/hooks/useHasScrolled.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Has Scrolled hook.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useWindowScroll } from 'react-use';
+
+/**
+ * Returns whether the user has scrolled the page and adds/remove the googlesitekit-plugin--has-scrolled class to the body.
+ *
+ * @since n.e.x.t
+ *
+ * @return {boolean} `true` if the user has scrolled the page, `false` otherwise.
+ */
+export const useHasScrolled = () => {
+	const { y } = useWindowScroll();
```

Comment:
Looks like we shouldn't use this hook, especially at a high-level as it causes any consuming components to be rerendered every time `y` changes, even when using `useEffect` as mentioned here and rerendering the dashboard apps adds an additional 30-40ms in the profiler for me.

![image](https://user-images.githubusercontent.com/1621608/150503466-27d4e506-fdbb-45b9-ad93-875560889e02.png)

Even an alternative using a listener on `scroll` is more performant, but the rerender of the whole dashboard is something we should avoid.

This gives more control over when a rerender will happen but still had the same expense in terms of the rerendering of the dashboard, just less often.

I found that putting this hook in an empty component that is rendered in the dashboard solved this problem although I would imagine there's probably a more idomatic way to do this in React to avoid triggering unnecessary rerenders in the consuming component. cc: @tofumatt ?

![image](https://user-images.githubusercontent.com/1621608/150504345-6afc031d-4bb1-4747-aac7-4059194d9ca1.png)
![image](https://user-images.githubusercontent.com/1621608/150504355-aac22a66-2c87-4c0e-a5eb-1468bfa44a17.png)

![image](https://user-images.githubusercontent.com/1621608/150503822-62dc15d6-4580-40f4-a9e0-5aa8f97de1d9.png)

Reply:
In my experience that's the idiomatic way, hah! In this case we have a component with side-effects and placing it on its own and outside of other component structures is the way-to-go. I've seen similar things done with stuff like `<SetupRequests />` or `<GetUserProfile />` components before.

It's a bit _odd_ to see but it's totally fine and I think makes sense. That'd be the approach I would've taken here as well, so I say let's go with that approach 👍🏻